src/python/pants/backend/python/providers/python_build_standalone/rules.py

```diff
@@ -69,6 +75,232 @@ class PBSPythonInfo(TypedDict):
+    if not version_and_tag:
+        return None, None
+
+    parts = version_and_tag.split("+", 1)
+    py_version: Version | None = None
+    pbs_release_tag: Version | None = None
+
+    if len(parts) >= 1:
+        try:
+            py_version = Version(parts[0])
+        except InvalidVersion:
+            raise ValueError(f"Version `{parts[0]}` is not a valid Python version.")
+
+    if len(parts) == 2:
+        try:
+            pbs_release_tag = Version(parts[1])
+        except InvalidVersion:
+            raise ValueError(f"PBS release tag `{parts[1]}` is not a valid version.")
+
+    return py_version, pbs_release_tag
+
+
+def _parse_pbs_url(url: str) -> tuple[Version, Version, Platform]:
+    parsed_url = urllib.parse.urlparse(urllib.parse.unquote(url))
+    base_path = posixpath.basename(parsed_url.path)
+
+    base_path_no_prefix = base_path.removeprefix("cpython-")
+    if base_path_no_prefix == base_path:
+        raise ValueError(
+            f"Unable to parse the provided URL since it does not have a cpython prefix as per the PBS naming convention: {url}"
+        )
+
+    base_path_parts = base_path_no_prefix.split("-", 1)
+    if len(base_path_parts) != 2:
+        raise ValueError(
+            f"Unable to parse the provided URL because it does not follow the PBS naming convention: {url}"
+        )
+
+    py_version, pbs_release_tag = _parse_py_version_and_pbs_release_tag(base_path_parts[0])
+    if not py_version or not pbs_release_tag:
+        raise ValueError(
+            "Unable to parse the Python version and PBS release tag from the provided URL "
+            f"because it does not follow the PBS naming convention: {url}"
+        )
+
+    platform: Platform
+    match base_path_parts[1].split("-"):
+        case ["x86_64", "unknown", "linux", "gnu", *_]:
+            platform = Platform.linux_x86_64
+        case ["aarch64", "unknown", "linux", "gnu", *_]:
+            platform = Platform.linux_arm64
+        case ["x86_64", "apple", "darwin", *_]:
+            platform = Platform.macos_x86_64
+        case ["aarch64", "apple", "darwin", *_]:
+            platform = Platform.macos_arm64
+        case _:
+            raise ValueError(
+                "Unable to parse the platfornm from the provided URL "
+                f"because it does not follow the PBS naming convention: {url}"
+            )
+
+    return py_version, pbs_release_tag, platform
+
+
+def _parse_from_three_fields(parts: Sequence[str], orig_value: str) -> _ParsedPBSPython:
+    assert len(parts) == 3
+    url, sha256, size = parts
+
+    try:
+        py_version, pbs_release_tag, platform = _parse_pbs_url(url)
+    except ValueError as e:
+        raise ExternalToolError(
+            f"While parsing the `[{PBSPythonProviderSubsystem.options_scope}].known_python_versions` option, "
+            f"the value `{orig_value}` could not be parsed because the URL does not follow the "
+            f"PBS naming convention: {e}"
```

Comment:
Given most the errors in `_parse_pbs_url` mention "does not follow the PBS naming convention", what do you think about reducing redundancy here:

```suggestion
            f"the value `{orig_value}` could not be parsed: {e}"
```